### DISCUSSION TITLE
Bike-sharing forecasting: a forecasting 30 minutes ahead without using 'progressive_val_score'

### DISCUSSION THREAD

**Discussion by TxusLopez** [online-ml/river#744] [Q&A]
Hi,

I'm using the `progressive_val_score` function to evaluate the predictions X minutes ahead in my dataset. This function manages this issue by using the `delay` and `moment` parameters. I'm not sure about the results I am obtaining. My features and labels are not shifted, so I expected the function to do that, by managing which features at time t should be trained with the corresponding label, etc. 

I would like to see how I can do that manually or traditionally without using the function. For example in the  Bike-sharing forecasting toy example. 

Thanks

**Comment by MaxHalford** [comment]
Hey! Sorry but what's the question?

I'm asking this in a tongue in cheek manner because this is a complex topic. Do you want to see in what order the data is processed by the model? Under the hood, `evaluate.progressive_val_score` uses the `stream.simulate_qa` function. This is the function that manages the data so as to process in the correct order.

**Reply by TxusLopez** [reply]
Thanks Max, I will check the `stream.simulate_qa` and I will try to simulate its operation for my own purposes. Thanks a lot.

**Reply by MaxHalford** [reply]
Feel free to shout out if you need a hand :)
